lib/src/html_input_stream.dart

```diff
@@ -205,14 +205,14 @@ class HtmlInputStream {
 
   /// Read one character from the stream or queue if available. Return
   /// EOF when EOF is reached.
-  String char() {
+  String? char() {
     if (_offset >= _chars.length) return eof;
     return _isSurrogatePair(_chars, _offset)
         ? String.fromCharCodes([_chars[_offset++], _chars[_offset++]])
         : String.fromCharCodes([_chars[_offset++]]);
```

Comment:
Hmm. `String.fromCharCode(_chars[_offset++])` would probably be more efficient than allocating an extra list here..

Reply:
that's a good call, I'll follow up with a separate PR. https://github.com/dart-lang/html/pull/141